src/ripple/ledger/ApplyView.h

```diff
@@ -195,22 +195,74 @@ class ApplyView
 
     //--------------------------------------------------------------------------
 
+   /** Add an entry to a directory
+
+        @param directory the base of the directory
+        @param key the entry to insert
+        @param strictOrder keep entries in order of insertion
+        @param describe callback to add required entries to a new page
+    */
+    /** @{ */
+    std::pair<bool, std::uint64_t>
+    dirInsert (
+        Keylet const& directory,
+        uint256 const& key,
+        bool strictOrder,
+        std::function<void(std::shared_ptr<SLE> const&)> describe);
+
+    std::pair<bool, std::uint64_t>
+    dirInsert (
+        Keylet const& directory,
+        Keylet const& key,
+        bool strictOrder,
+        std::function<void(std::shared_ptr<SLE> const&)> describe);
+    /** @} */
+
+    /** Removes an entry from a directory
+
+        @param directory the base of the directory
+        @param page the page number for this page
+        @param key the entry to remove
+        @param keepRoot if deleting the last entry, don't
+                        delete the directory itself.
+
+        Effects:
+            Removes one or more pages, potentially removing
+            the directory itself.
+    */
+    /** @{ */
+    bool
+    dirRemove (
+        Keylet const& directory,
+        std::uint64_t page,
+        uint256 const& key,
+        bool keepRoot);
+
+    bool
+    dirRemove (
+        Keylet const& directory,
+        std::uint64_t page,
+        Keylet const& key,
```

Comment:
One takes a `Keylet` but Sometimes we don't have a `Keylet` available, so we offer the other version which accepts a `uint256`. We could have required callers to construct the `Keylet` but I wanted to minimize changes for this round. 

I will be removing uses of the old API along with the amendment to enable sorting of owner directories, so that will go away.